### PR TITLE
Fixes oldcoder stuff with radio controllers

### PR DESCRIPTION
**Pull request by ShiftyRail**
Mostly moving things from `New()` to `initialize()`.

### CODE REVIEW

**Comment by DamianX** [comment]
Atmos is weird with regards to `initialize()`. There's lots of atmos code that calls it, and some of it expects a return value and most atmos implementations of it perform non-trivial work. What you did here is probably OK in the sense that it shouldn't break anything but it's not optimal.
In my attempted initialize rework I moved the current logic into a new proc called `atmos_init()`, made atmos devices call that, then added the frequency stuff to the actual `initialize()` like you did here.
Do what you will with this information https://github.com/vgstation-coders/vgstation13/pull/23076/commits/e98cddf941a6913d90c5de32455adce421c5a1d5

File: code/ATMOSPHERICS/components/trinary_devices/filter.dm [hunk 1]
```diff
@@ -26,18 +26,18 @@ Filter types:
...
+/obj/machinery/atmospherics/trinary/filter/proc/set_frequency(new_frequency)
 	radio_controller.remove_object(src, frequency)
 	frequency = new_frequency
 	if(frequency)
 		radio_connection = radio_controller.add_object(src, frequency, RADIO_ATMOSIA)
 
-obj/machinery/atmospherics/trinary/filter/New()
-	if(radio_controller)
+/obj/machinery/atmospherics/trinary/filter/New()
+	if(ticker && ticker.mode == GAME_STATE_PLAYING)
```
Comment (jknpj) [inline code/ATMOSPHERICS/components/trinary_devices/filter.dm:36]: why not simply move that code from initialize to new()?

Comment (DamianX) [reply code/ATMOSPHERICS/components/trinary_devices/filter.dm:36]: Because new gets called before radio_controller exists

Comment (jknpj) [reply code/ATMOSPHERICS/components/trinary_devices/filter.dm:36]: understandable

**Comment by DamianX** [comment]
Is it possible this made it so radios start with the wrong frequency? Because it sure looks like it's happening

**Pull request merged**